web/packages/url-utils/src/index.ts

```diff
@@ -0,0 +1,91 @@
+import { sha256 } from 'js-sha256';
+
+const AVATAR_BASE_URL = 'https://www.gravatar.com/avatar/';
+const PROFILE_BASE_URL = 'https://www.gravatar.com/';
+
+// Gravatar profile formats
+export enum GravatarFormat {
+	HTML = '',
+	JSON = 'json',
+	XML = 'xml',
+	PHP = 'php',
+	VCF = 'vcf',
+	QR = 'qr',
+}
+
+// Gravatar default images
+export enum GravatarDefault {
+	STATUS_404 = '404',
+	MYSTERY_PERSON = 'mp',
+	IDENTICON = 'identicon',
+	MONSTERID = 'monsterid',
+	WAVATAR = 'wavatar',
+	RETRO = 'retro',
+	ROBOHASH = 'robohash',
+	TRANSPARENT_PNG = 'blank',
+}
+
+// Gravatar ratings
+export enum GravatarRating {
+	GENERAL = 'g',
+	PARENTAL_GUIDANCE = 'pg',
+	RESTRICTED = 'r',
+	X = 'x',
+}
+
+interface GravatarAvatarOptions {
+	default?: GravatarDefault | string;
+	size?: number;
+	rating?: GravatarRating;
+	forceDefault?: boolean;
+}
+
+// Function to get the SHA256 hash of the email
+function getHash( email: string ): string {
+	if ( typeof email !== 'string' || email.trim() === '' ) {
+		throw new Error( 'Valid email is required' );
+	}
+	return sha256( email.trim().toLowerCase() );
+}
+
+// Function to generate the query string from options
+function getQueryString( options: GravatarAvatarOptions ): string {
+	const params: { [ key: string ]: string } = {};
+	for ( const key in options ) {
+		if ( key === 'size' ) {
+			params.size = options.size.toString();
```

Comment:
Error handling might be needed to prevent JavaScript errors (e.g., `{}.toString()`), especially for non-TypeScript developers.

```suggestion
			params.size = options.size?.toString();
```